### PR TITLE
[Merged by Bors] - CI fix: move download web3signer binary out of build script

### LINKED ISSUE #3812: Rate limit for web3signer_tests on CI
## Description

We're occasionally getting rate-limited calling the Github API to download the `web3signer_tests`. E.g. https://github.com/sigp/lighthouse/actions/runs/3708544134/jobs/6286220721

The code triggering the rate limit is:

https://github.com/sigp/lighthouse/blob/bf533c8e42cc73c35730e285c21df8add0195369/testing/web3signer_tests/build.rs#L35-L45

Running that API call in a loop in my terminal I can trigger a 403 response by making more than 60 requests/second:

```json
{"message":"API rate limit exceeded for x.x.x.x. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
```

## Steps to resolve

I think the best option would be to plumb the `GITHUB_TOKEN` into that build script so that it can use it to bypass the rate limit. The build script could read an _optional_ secret from its env when it runs and add this to the request headers (when running locally the header would not be added).

I think if we do something like `echo ${{ secrets.GITHUB_TOKEN }} >> $GITHUB_ENV` that _might_ be OK, as long as Github scrubs the token from the logs. Tokens are unique to each CI run and have very limited `read` permissions on pull requests. See: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

### PR DESCRIPTION
**Pull request by jimmygchen**
## Issue Addressed

Attempt to fix #3812 

## Proposed Changes

Move web3signer binary download script out of build script to avoid downloading unless necessary. If this works, it should also reduce the build time for all jobs that runs compilation.

### CODE REVIEW

**Comment by jimmygchen** [comment]
@michaelsproul I've moved the download out of the build script to the tests, and it seems to be working! There's one job failure due to port conflicts in the eth1 simulator tests, but I don't think it's related to the change.

File: testing/web3signer_tests/src/lib.rs [hunk 1]
```diff
@@ -10,15 +10,20 @@
 //! - The signatures generated by Web3Signer are identical to those which Lighthouse generates.
 //!
 //! There is a build script in this crate which obtains the latest version of Web3Signer and makes
```
Comment (michaelsproul) [inline testing/web3signer_tests/src/lib.rs]: This comment is no longer accurate, could mention `get_web3signer` instead

**Review by michaelsproul — APPROVED** [review]
Looks great! Thanks for taking the initiative on this!

File: testing/web3signer_tests/src/lib.rs (hunk 1 above)
Comment (jimmygchen) [reply testing/web3signer_tests/src/lib.rs]: Good catch!

Comment (jimmygchen) [reply testing/web3signer_tests/src/lib.rs]: Fixed

**Comment by michaelsproul** [comment]
Nice! Let's go!

bors r+

**Pull request closed without merging**